### PR TITLE
Upgrade webpack to version 4

### PR DESCRIPTION
**Pull request by anler**
The current version of Webpack is 4 and we are using 3. This commit
upgrades our configurations to version 4. Note some dependencies are
still in alpha stage, but by the time we release sheikah, for sure those
depencencies will be already stable.

# PR Prelude

Thank you for contributing to sheikah! :)

**Please fill in this template (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [X] I have read and understood [CODE_OF_CONDUCT][code] document.
- [X] I have read and understood [CONTRIBUTING][cont] document.
- [X] I have read and understood [docs/styleguide][style] document.
- [X] I have included tests for the changes in my PR. If not, I have included a
  rationale for why I haven't.
- [X] My code is formatted and is accepted by `yarn fmt-verify`.
- [X] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful

[Please explain **in detail** why the changes in this PR are needed.]

It is not **strictly** necessary but it is usually better to upgrade versions one by one rather than skip several updates and have more configuration to migrate.

Also, I'm adding a short commit that moves some dev dependencies into `devDependencies` in `package.json`

[code]: https://github.com/witnet/sheikah/blob/master/docs/CODE_OF_CONDUCT.md
[cont]: https://github.com/witnet/sheikah/blob/master/docs/CONTRIBUTING.md
[style]: https://github.com/witnet/sheikah/blob/master/docs/STYLEGUIDE.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -123,6 +123,7 @@
     "babel-preset-react": "^6.24.1",
     "babel-preset-typescript": "^7.0.0-alpha.19",
     "boiler-room-custodian": "^0.6.2",
+    "commitlint": "^6.2.0",
```
Comment (aesedepece) [inline package.json]: Related to #76

**Comment by aesedepece** [comment]
Cool! @kronolynx struggled to upgrade to Webpack 4 but it turned to be an authentic nightmare.

E2E tests are not passing for me.

**Comment by anler** [comment]
@aesedepece I had the same issue, try running `yarn clean` before running the e2e tests.

**Comment by aesedepece** [comment]
:clap: :clap: :clap:

**Pull request merged**